ibis/pyspark/compiler.py

```diff
@@ -1111,53 +1092,73 @@ def compile_window_op(t, expr, scope, timecontext, **kwargs):
         else:
             pyspark_window = pyspark_window.rowsBetween(start, end)
 
-    result = t.translate(
-        operand, scope, timecontext, window=pyspark_window, context=context
+    res_op = operand.op()
+
+    # For NotAll and NotAny, negation must be applied after .over(window)
+    # Here we rewrite node to be its negation, and negate it back after
+    # translation and window operation
+    negated = False
```

Comment:
Let's have cleaner branching like this:

```
if isinstance(op, (ops.NotAll, ops.NotAny)):
   return ...
elif isinstance(op, (ops.MinRank,ops.DenseRank, ops.RowNumber)):
    return ...
else:
    return ...
```

Reply:
That's indeed more readable. However, I argue that this may bring duplication in code:
```
result = t.translate(operand, scope, timecontext, context=context).over(
        pyspark_window
    )
```
will repeat 3 times.
The motivation for the current structure is pre_processing, translation, post_processing.

Reply:
I revised as you suggested for more readability